### PR TITLE
HACKING, packaging: update dependencies and information on installing development dependencies

### PR DESCRIPTION
**Pull request by bboozzoo**
Update instructions on installing development dependencies. Include an example for Fedora.

### CODE REVIEW

**Review by olivercalder — COMMENTED** [review]
With these instructions executed, `go test ./...` is able to run, but there are many test failures. I tried both without and with snapd installed on the system.

Without snapd: https://termbin.com/rw4l
With snapd: https://termbin.com/fhfb

Not sure if this means there are other dependencies required for tests to pass, or if this means tests make incorrect assumptions about the systems on which they run (and thus should be fixed). I'll investigate a bit more.

**Comment by olivercalder** [comment]
Had a peek with Zygmunt... looks like `mksquashfs` is adding security xattrs which then denies permission to unsquashfs. I'll try to find where `mksquashfs` is used in tests and see if I can add `--no-xattrs` when using `mksquashfs` in tests.

Also, there are at least two other classes of problems I've spotted:

- Mount count incorrect (probably related to failed unsquashfs?)
- Something actually relevant:

```
In file included from /usr/include/features.h:527,
                 from /usr/include/errno.h:25,
                 from /tmp/check-1158390419/1/seccomp_syscall_runner.c:3:
/usr/include/gnu/stubs.h:7:11: fatal error: gnu/stubs-32.h: No such file or directory
    7 | # include <gnu/stubs-32.h>
      |           ^~~~~~~~~~~~~~~~
compilation terminated.
```

- Another:

```
FAIL: main_test.go:876: snapSeccompSuite.TestCompatArchWorks

main_test.go:909:
    s.runBpf(c, t.seccompAllowlist, t.bpfInput, t.expected)
main_test.go:378:
    c.Assert(exitCode == 0 || exitCode == 10 || exitCode == 20, Equals, true, Commentf("unexpected exit code: %v for %v - test setup broken", exitCode, seccompAllowlist))
... obtained bool = false
... expected bool = true
... unexpected exit code: -1 for read - test setup broken
```

- Another:

```
FAIL: image_test.go:3698: imageSuite.TestSetupSeedCore20DelegatedSnapAssertionMaxFormatsHappy

image_test.go:3735:
    c.Check(err, IsNil)
... value *errors.errorString = &errors.errorString{s:"cannot run unsquashfs: \n-----\nwrite_xattr: could not write xattr security.selinux for file /tmp/read-file2310665016/unpack/meta/snap.yaml because you're not superuser!\nwrite_xattr: to avoid this error message, either specify -xattrs-include '^user.', -no-xattrs, or run as superuser!\nFurther error messages of this type are suppressed!\nParallel unsquashfs: Using 20 processors\n1 inodes (1 blocks) to write\n\n/tmp/read-file2310665016/unpack\n/tmp/read-file2310665016/unpack/meta\n/tmp/read-file2310665016/unpack/meta/snap.yaml\n\ncreated 1 file\ncreated 2 directories\ncreated 0 symlinks\ncreated 0 devices\ncreated 0 fifos\ncreated 0 sockets\ncreated 0 hardlinks\n-----"} ("cannot run unsquashfs: \n-----\nwrite_xattr: could not write xattr security.selinux for file /tmp/read-file2310665016/unpack/meta/snap.yaml because you're not superuser!\nwrite_xattr: to avoid this error message, either specify -xattrs-include '^user.', -no-xattrs, or run as superuser!\nFurther error messages of this type are suppressed!\nParallel unsquashfs: Using 20 processors\n1 inodes (1 blocks) to write\n\n/tmp/read-file2310665016/unpack\n/tmp/read-file2310665016/unpack/meta\n/tmp/read-file2310665016/unpack/meta/snap.yaml\n\ncreated 1 file\ncreated 2 directories\ncreated 0 symlinks\ncreated 0 devices\ncreated 0 fifos\ncreated 0 sockets\ncreated 0 hardlinks\n-----")

image_test.go:3758:
    c.Check(later, DeepEquals, map[string]bool{
        s.AssertedSnapID("snapd"):      true,
        s.AssertedSnapID("core20"):     true,
        s.AssertedSnapID("pc-kernel"):  true,
        s.AssertedSnapID("pc"):         true,
        s.AssertedSnapID("required20"): true,
    })
... obtained map[string]bool = map[string]bool{"DLqre5XGLbDqg9jPtiAhRRjDuPVa5X1q":true, "PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4":true, "pckernelidididididididididididid":true}
... expected map[string]bool = map[string]bool{"DLqre5XGLbDqg9jPtiAhRRjDuPVa5X1q":true, "PMrrV4ml8uWuEUDBT8dSGnKUYbevVhc4":true, "pcididididididididididididididid":true, "pckernelidididididididididididid":true, "required20ididididididididididid":true}
... Difference:
...     ["pcididididididididididididididid"]: (missing) != %!q(bool=true)
...     ["required20ididididididididididid"]: (missing) != %!q(bool=true)
```

...etc.

**Review by olivercalder — APPROVED** [review]
Thanks!

**Comment by bboozzoo** [comment]
Added a blocked label, the multilib dependencies expressed in the RPM spec seem to make rpmbuild quite unhappy, so I need to figure out what's going on there.

**Comment by bboozzoo** [comment]
I have dropped the changes to snapd.spec as they were incorrect. So we're left with a note in HACKING.md and little tweak to the spread test prepare.

File: HACKING.md [hunk 1]
```diff
@@ -46,10 +46,15 @@ the `snapd` project, please see [Contributing to snapd](./CONTRIBUTING.md).
...
+    sudo apt build-dep .
 
 Package build dependencies for other distributions can be found under the
-[./packaging/](./packaging/) directory.
+[./packaging/](./packaging/) directory. Eg. for Fedora use:
+
+    cd packaging/fedora
+    sudo dnf install -y rpmdevtools
+    sudo dnf install -y $(rpmspec -q --buildrequires snapd.spec)
+    sudo dnf install glibc-static.i686 glibc-devel.i686
```
Comment (olivercalder) [inline HACKING.md:57]: Are these required for unit tests to pass but not for building the snapd rpm? Else why not include in snapd.spec?

**Review by olivercalder — APPROVED** [review]
Thanks!

File: HACKING.md (hunk 1 above)
Comment (bboozzoo) [reply HACKING.md:57]: That's the problem, I can't properly include them in the spec without breaking something. I've actually had a patch adding then to BuildRequires for x86_64 arch, but rpmbuild got unhappy and complained, even though dependencies were satisfied according to dnf, so I dropped the patch. Those packages are already included in the build images we use here and in Fedora infra.

**Pull request merged**